ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoder.scala

```diff
@@ -43,7 +43,8 @@ class H2OTargetEncoder(override val uid: String)
       "H2OContext needs to be created in order to use target encoding. Please create one as H2OContext.getOrCreate().")
     val input = h2oContext.asH2OFrame(dataset.toDF())
     val distinctInputCols = getInputCols().flatten.distinct
-    input.convertColumnsToCategorical(distinctInputCols)
+    val toCategorical = if (getProblemType() == "Regression") distinctInputCols else distinctInputCols ++ Seq(getLabelCol())
```

Comment:
shouldn't this be case insensitive, spark ml accepts eg. "family" = "multinomial"

Reply:
the [setter method](https://github.com/h2oai/sparkling-water/pull/2462/files#diff-cfe2d864665ec1d4b957eab0d39ad7117e6af6a2728d509270caee83b8449475R102) and python -> scala parameter serializer should convert any string value into this capitalized form, but I will use the lowercase form in test to be 100% sure.

Reply:
I didn't know that, don't worry then! thank you for explaining